can/interfaces/neovi_api/neovi_api.py

```diff
@@ -0,0 +1,172 @@
+"""
+pyneovi interface module.
+
+pyneovi is a Python wrapper around the API provided by Intrepid Control Systems
+for communicating with their NeoVI range of devices.
+
+Implementation references:
+* http://pyneovi.readthedocs.io/en/latest/
+* https://bitbucket.org/Kemp_J/pyneovi
+"""
+
+import logging
+
+import sys
+
+logger = logging.getLogger(__name__)
+
+try:
+    from queue import Queue, Empty
+except ImportError:
+    from Queue import Queue, Empty
+
+try:
+    from neovi import neodevice
+    from neovi import neovi
+    from neovi.structures import icsSpyMessage
+except ImportError:
+    print
+
+if sys.platform == "win32":
+    try:
+        from neovi import neodevice
+        from neovi import neovi
+        from neovi.structures import icsSpyMessage
+    except ImportError as e:
+        logger.warning("Cannot load pyneovi: %s", e)
+else:
+    # Will not work on other systems, but have it importable anyway for
+    # tests/sphinx
+    logger.warning("pyneovi library does not work on %s platform", sys.platform)
+
+from can import Message
+from can.bus import BusABC
+
+
+def neo_device_name(device_type):
+    names = {
+        neovi.NEODEVICE_BLUE: 'neoVI BLUE',
+        neovi.NEODEVICE_DW_VCAN: 'ValueCAN',
+        neovi.NEODEVICE_FIRE: 'neoVI FIRE',
+        neovi.NEODEVICE_VCAN3: 'ValueCAN3',
+        neovi.NEODEVICE_YELLOW: 'neoVI YELLOW',
+        neovi.NEODEVICE_RED: 'neoVI RED',
+        neovi.NEODEVICE_ECU: 'neoECU',
+        # neovi.NEODEVICE_IEVB: ''
+    }
+    return names.get(device_type, 'Unknown neoVI')
+
+
+class NeoVIBus(BusABC):
+    """
+    The CAN Bus implemented for the pyneovi interface.
+    """
+
+    def __init__(self, channel=None, can_filters=None, **config):
```

Comment:
Can we leave that for a future improvement for now? I do not have a ms can bus to test at this point.